platform/cli/cli.ts

```diff
@@ -40,15 +41,25 @@ const run = (argv: { [key: string]: any }) => {
     .map(rawConfig => merge({}, rawConfig, configOverrides));
 
   const root = new Root(rawConfig$, env, onShutdown);
-  root.start();
+
+  if (argv.kbnServer) {
+    const proxy = (argv.kbnServer.proxy = new Proxy(
+      root.logger.get('proxy'),
+      argv.kbnServer
+    ));
+    proxy.on('platform-start', async () => await root.start());
```

Comment:
I'm just exploring the idea, but maybe it's not that bad if we start `root` only when old platform starts Hapi with our `proxy` as a listener. This way we reuse port that old platform supplies us via `listener` so we don't have to read `kbnServer.settings.server.port`...